tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -0,0 +1,174 @@
+log = logging.getLogger(__name__)
+TEST_PROJECT = 'test-project'
+PROJECT = ocp.OCP(kind='Project', namespace=TEST_PROJECT)
+
+
+@pytest.fixture()
+def test_fixture(request):
+    """
+    Setup and teardown
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Create new project
+    Create PVCs
+    """
+    # Create new project
+    assert PROJECT.new_project(TEST_PROJECT), (
+        f'Failed to create new project {TEST_PROJECT}'
+    )
+
+    # Parameters for PVC yaml as dict
+    pvc_data = defaults.CSI_PVC_DICT.copy()
+    pvc_data['metadata']['namespace'] = TEST_PROJECT
+    pvc_data['spec']['storageClassName'] = self.sc_obj.name
+    pvc_data['metadata']['name'] = self.pvc_base_name
+
+    # Create 100 PVCs
+    pvc_objs = create_multiple_pvc(self.number_of_pvc, pvc_data)
+    log.info(f'Created initial {self.number_of_pvc} PVCs')
+    self.pvc_objs_initial = pvc_objs[:]
+
+    # Verify PVCs are Bound and fetch PV names
+    for pvc in self.pvc_objs_initial:
+        pvc.reload()
+        assert pvc.status == constants.STATUS_BOUND, (
+            f'PVC {pvc.name} is not Bound'
+        )
+        self.initial_pvs.append(pvc.backed_pv)
+    log.info(f'Initial {self.number_of_pvc} PVCs are in Bound state')
+
+
+def teardown(self):
+    """
+    Delete PVCs
+    Delete project
+    """
+    # Delete newly created PVCs
+    command = (
+        f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
+        f'{self.pvc_base_name_new}$i -n {TEST_PROJECT};done'
+    )
+    proc = run_async(command)
+    assert proc, (
+        f'Failed to execute command for deleting {self.number_of_pvc} PVCs'
+    )
+
+    # Verify command to delete PVCs
+    ret, _, _ = proc.async_communicate()
+    assert not ret, 'Deletion of newly created PVCs failed'
+    log.info(f'Newly created {self.number_of_pvc} PVCs are now deleted.')
+
+    # Switch to default project
+    ret = ocp.switch_to_default_rook_cluster_project()
+    assert ret, 'Failed to switch to default rook cluster project'
+
+    # Delete project created for the test case
+    PROJECT.delete(resource_name=TEST_PROJECT)
+
+
+@tier1
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__,
+    test_fixture.__name__
+)
+class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
+    """
+    Test to verify concurrent creation and deletion of multiple PVCs
+    """
+    number_of_pvc = 100
+    pvc_base_name = 'test-pvc'
+    pvc_base_name_new = 'test-pvc-re'
+    initial_pvs = []
+    pvc_objs_initial = []
+    pvc_objs_new = []
+
+    def test_multiple_pvc_concurrent_creation_deletion(self):
+        """
+        To exercise resource creation and deletion
+        """
+        # Start deleting 100 PVCs
+        command = (
+            f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
```

Comment:
are you using oc command directly here?  we have pvc class now, can you move the functionality there?

Reply:
@vasukulkarni
The intention here is to issue a command for deleting PVCs and immediately start creating another set of PVCs so that both process will run simultaneously. The command to delete PVCs will be verified in the last part of the test case.